xclim/testing/tests/test_generic.py

```diff
@@ -142,3 +143,39 @@ def test_doyminmax(q_series):
             assert attr in da.attrs.keys()
         assert da.attrs["units"] == ""
         assert da.attrs["is_dayofyear"] == 1
+
+
+def test_aggregate_between_dates():
```

Comment:
As said above, please add a test where time axes of `data` and  `start` or `end` do not entirely overlap.
Also, I think some tests for further edgecases could be nice : 
- `data` and `start` not in the same calendar
- `start` and `end` not at the same freq (test if the error is the good one)
- Test other frequencies (QS-DEC, for instance)